### PR TITLE
[TIMOB-25233] Implement TextField.setSelection

### PR DESCRIPTION
**Pull request by infosia**
[TIMOB-25233](https://jira.appcelerator.org/browse/TIMOB-25233)

```js
var _window = Ti.UI.createWindow({ backgroundColor: 'green' });
var textfield0 = Titanium.UI.createTextField({
    width: Ti.UI.FILL,
    top: 100,
    value: 'textfield0'
});

_window.addEventListener('open', function () {
    textfield0.setSelection(1, 4);
});

_window.add(textfield0);
_window.open();
```

Expected: texts between 1 to 3 character (`'ext'`) is selected.

### CODE REVIEW

File: Source/UI/src/TextField.cpp [hunk 1]
```diff
@@ -295,6 +295,19 @@ namespace TitaniumWindows
 			return false;
 		}
 
+		void TextField::setSelection(const std::uint32_t& start, const std::uint32_t& end) TITANIUM_NOEXCEPT
+		{
+			if (text_box__) {
+				text_box__->Select(start, (end - start));
```
Comment (garymathews) [inline Source/UI/src/TextField.cpp]: Should probably validate `start < length` and `end < length`

Comment (infosia) [reply Source/UI/src/TextField.cpp]: Yes, updated 👍

**Comment by ssjsamir** [comment]
@infosia 'ext' is not highlighted on a mobile device running windows 10  (10.0.14393.0) but when running on ws-local 'ext' is highlighted as expected.

**Comment by infosia** [comment]
@ssjsamir Hmm I think it is platform component issue. Microsoft doesn't say it does not work on mobile but it does not actually  work :(  I will add note to our API doc.

**Pull request merged**